frontends/smvparser.y

```diff
@@ -446,14 +564,23 @@ basic_expr: constant {
                   smt::Term res = enc.solver_->make_term(smt::Ge, a->getTerm(), b->getTerm());
                   $$ = new SMVnode(res);
               }else{
-                  smt::Term res = enc.solver_->make_term(smt::BVUge, a->getTerm(), b->getTerm());
-                  $$ = new SMVnode(res);
+                  SMVnode::BVtype bvs_a = a->getBVType();
+                  SMVnode::BVtype bvs_b = b->getBVType();
+                  if (bvs_a == bvs_b == SMVnode::Unsigned){
+                    smt::Term res = enc.solver_->make_term(smt::BVUge, a->getTerm(), b->getTerm());
+                    $$ = new SMVnode(res);
+                  } else if (bvs_a == bvs_b == SMVnode::Signed){
+                    smt::Term res = enc.solver_->make_term(smt::BVSge, a->getTerm(), b->getTerm());
+                    $$ = new SMVnode(res);
+                  } else{
+                    throw CosaException ("Unsigned/Signed unmatch");
+                  } 
               }
             }
             | OP_MINUS basic_expr %prec UMINUS{
```

Comment:
This is just for me to understand. What is the `%prec` here?

Reply:
I followed bison's manual to use %prec for unary minus in the grammar

"The %prec modifier declares the precedence of a particular rule by specifying a terminal symbol whose precedence should be used for that rule."

Reply:
Thanks!